Fixed missing imports and missing libs

maximilien-noal opened the pull request:
Now it builds !
I still have asserts problems at runtime, but that's another topic.

maximilien-noal committed:
Linux: Fixed missing imports when building

Signed-off-by: Maximilien Noal <noal.maximilien@gmail.com>

maximilien-noal committed:
Fixed missing GLX when linking

Signed-off-by: Maximilien Noal <noal.maximilien@gmail.com>

maximilien-noal committed:
git: ignore build/ dir

Signed-off-by: Maximilien Noal <noal.maximilien@gmail.com>

madmoose commented on CMakeLists.txt:
```diff
@@ -17,7 +17,9 @@ set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
 target_include_directories(chani PRIVATE 3rdparty/glfw/include/)
 target_sources(chani PRIVATE 3rdparty/imgui/examples/libs/gl3w/GL/gl3w.c)
-target_link_libraries(chani glfw)
+set(OpenGL_GL_PREFERENCE GLVND)
```
Can you put these in a conditional block for Linux (and/or WSL)? Like we have for Mac OS X further down. They break the CMake build on Mac OS X.

maximilien-noal replied:
Will do !

maximilien-noal replied:
Update done.

maximilien-noal committed:
CMakeList.txt : for macOS, Linux, and Windows

Windows won't build yet because there is no unistd there. Among other nix things.
